adapters/unruly/unruly.go

```diff
@@ -12,137 +12,148 @@ import (
-			Message: fmt.Sprintf("Unexpected status code: %d. Run with request.debug = 1 for more info", response.StatusCode),
-		}
+		return nil, []error{&errortypes.BadServerResponse{
+			Message: fmt.Sprintf("unexpected status code: %d. Run with request.debug = 1 for more info", response.StatusCode),
+		}}
 	}
-	return nil
-}
-
-func convertToAdapterBidResponse(response *adapters.ResponseData, internalRequest *openrtb2.BidRequest) (*adapters.BidderResponse, []error) {
-	var errs []error
 	var bidResp openrtb2.BidResponse
 	if err := json.Unmarshal(response.Body, &bidResp); err != nil {
-		return nil, []error{err}
+		return nil, []error{&errortypes.BadServerResponse{
+			Message: fmt.Sprintf("bad server response: %d. ", err),
+		}}
 	}
+
 	bidResponse := adapters.NewBidderResponseWithBidsCapacity(5)
+
+	var errs []error
 	for _, sb := range bidResp.SeatBid {
 		for i := range sb.Bid {
-			bidType, err := getMediaTypeForImpWithId(sb.Bid[i].ImpID, internalRequest.Imp)
+			var bidType, err = getMediaTypeForImp(sb.Bid[i].ImpID, internalRequest.Imp)
 			if err != nil {
-				errs = append(errs, err)
+				errs = append(errs, err...)
 			} else {
-				b := &adapters.TypedBid{
+				bidResponse.Bids = append(bidResponse.Bids, &adapters.TypedBid{
 					Bid:     &sb.Bid[i],
 					BidType: bidType,
-				}
-				bidResponse.Bids = append(bidResponse.Bids, b)
+				})
 			}
 		}
 	}
+
 	return bidResponse, errs
 }
 
-func convertBidderNameInExt(imp *openrtb2.Imp) (*openrtb2.Imp, error) {
-	var bidderExt adapters.ExtImpBidder
-	if err := json.Unmarshal(imp.Ext, &bidderExt); err != nil {
-		return nil, err
-	}
-	var unrulyExt openrtb_ext.ExtImpUnruly
-	if err := json.Unmarshal(bidderExt.Bidder, &unrulyExt); err != nil {
-		return nil, err
-	}
-	var impExtUnruly = ImpExtUnruly{Unruly: openrtb_ext.ExtImpUnruly{
-		SiteID: unrulyExt.SiteID,
-		UUID:   unrulyExt.UUID,
-	}}
-	bytes, err := json.Marshal(impExtUnruly)
-	if err != nil {
-		return nil, err
+func getMediaTypeForImp(impId string, imps []openrtb2.Imp) (openrtb_ext.BidType, []error) {
+	var errs []error
+	var noMatchingImps []string
+	mediaType := openrtb_ext.BidTypeBanner
+	for _, imp := range imps {
+		if imp.ID == impId {
+			if imp.Banner != nil {
+				mediaType = openrtb_ext.BidTypeBanner
+			} else if imp.Video != nil {
+				mediaType = openrtb_ext.BidTypeVideo
+			} else {
+				errs = append(errs, fmt.Errorf("bid responses mediaType didn't match supported mediaTypes"))
+			}
+			return mediaType, nil
+		} else {
+			noMatchingImps = append(noMatchingImps, imp.ID)
+		}
 	}
-	imp.Ext = bytes
-	return imp, nil
-}
 
-func (a *UnrulyAdapter) MakeBids(internalRequest *openrtb2.BidRequest, externalRequest *adapters.RequestData, response *adapters.ResponseData) (*adapters.BidderResponse, []error) {
-	if err := CheckResponse(response); err != nil {
-		return nil, []error{err}
+	if len(noMatchingImps) > 0 {
+		errs = append(errs, fmt.Errorf("the following imps id's %v didn't match impId %s", noMatchingImps, impId))
```

Comment:
Perhaps this error should be a little more generic and not include a variable name like `impId` in it. How about something like: "Bid response imp ID %s not found in bid request containing imps %v\n"?

Reply:
done